pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -286,3 +295,218 @@ func filterInformerActions(actions []core.Action) []core.Action {
 
 	return ret
 }
+
+// TestOverlappingDeployments ensures that an overlapping deployment will not be synced by
+// the controller.
+func TestOverlappingDeployments(t *testing.T) {
+	f := newFixture(t)
+	now := metav1.Now()
+	later := metav1.Time{Time: now.Add(time.Minute)}
+
+	foo := newDeployment("foo", 1, nil, nil, nil, map[string]string{"foo": "bar"})
+	foo.CreationTimestamp = now
+	bar := newDeployment("bar", 1, nil, nil, nil, map[string]string{"foo": "bar"})
+	bar.CreationTimestamp = later
+
+	f.dLister = append(f.dLister, foo, bar)
+	f.objects = append(f.objects, foo, bar)
+
+	f.expectUpdateDeploymentStatusAction(bar)
+	f.run(getKey(bar, t))
+
+	actions := f.client.Actions()
+	d := actions[0].(core.UpdateAction).GetObject().(*extensions.Deployment)
+	if len(d.Annotations[util.OverlapAnnotation]) == 0 {
+		t.Errorf("annotations werent updated for the overlapping deployment: %v", d.Annotations)
```

Comment:
ok for checking the content - not sure where else the function is going to be reused so I am leaving it out for now.